server/src/http/broadcaster.rs

```diff
@@ -221,8 +251,134 @@ impl Broadcaster {
         // disconnected clients will get swept up by `remove_stale_clients`
         let send_events = client_events
             .iter()
-            .map(|(client, event)| client.send(event.clone()));
+            .map(|(ClientData { sender, .. }, event)| sender.send(event.clone()));
 
         let _ = future::join_all(send_events).await;
     }
 }
+
+fn project_filter(projects: Vec<String>) -> FeatureFilter {
+    Box::new(move |feature| {
+        if let Some(feature_project) = &feature.project {
+            projects.is_empty()
+                || projects.contains(&"*".to_string())
+                || projects.contains(feature_project)
+        } else {
+            false
+        }
+    })
+}
+
+#[cfg(test)]
+mod test {
```

Comment:
I put this test in here because it tests the broadcaster, but I couldn't find a way to do it without the 1s timeout later. Waiting in unit tests isn't great, but it feels like it might be a fine tradeoff.